openedx/core/djangoapps/safe_sessions/tests/test_utils.py

```diff
@@ -118,14 +130,20 @@ def assert_request_user_mismatch(self, user_at_request, user_at_response):
             yield
 
     @contextmanager
-    def assert_session_user_mismatch(self, user_at_request, user_in_session):
-        """
-        Asserts that the logger was called when request.user at request
-        time doesn't match the request.user at response time.
-        """
-        with self.assert_logged(
-            "SafeCookieData user at request '{}' does not match user in session: '{}'".format(
-                user_at_request, user_in_session
-            ),
+    def assert_request_and_session_user_mismatch(self, user_at_request, user):
```

Comment:
And if I need to make them separate then I will have to introduce new manager like:
```
    @contextmanager
    def assert_logged_with_message(self, log_string, log_level='error'):
        with patch('openedx.core.djangoapps.safe_sessions.middleware.log.' + log_level) as mock_log:
            yield
            mock_log.assert_any_call(log_string)
```
and use it in both managers `assert_request_user_mismatch` and `assert_session_user_mismatch` like:

``` 
    @contextmanager
    def assert_request_user_mismatch(self, user_at_request, user_at_response):
        """
        Asserts that the logger was called when request.user at request
        time doesn't match the request.user at response time.
        """
        with self.assert_logged_with_message(...):
            yield

    @contextmanager
    def assert_session_user_mismatch(self, user_at_request, user_in_session):
        """
        Asserts that the logger was called when request.user at request
        time doesn't match the request.user at response time.
        """
        with self.assert_logged_with_message(...):
            yield
```
And finally `test_different_user_at_step_2_error` will be like:
```
def test_different_user_at_step_2_error(self):
        self.request.safe_cookie_verified_user_id = "different_user"

        with self.assert_request_user_mismatch("different_user", self.user.id):
            self.assert_response(set_request_user=True, set_session_cookie=True)

        with self.assert_session_user_mismatch("different_user", self.user.id):
            self.assert_response(set_request_user=True, set_session_cookie=True)
```

@nasthagiri What do you think?